src/datarobotx/idp/calendars.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# DataRobot, Inc.
+#
+# This is proprietary source code of DataRobot, Inc. and its
+# affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+# https://www.datarobot.com/wp-content/uploads/2021/07/DataRobot-Tool-and-Utility-Agreement.pdf
+
+
+import datetime as dt
+from pathlib import Path
+from typing import Union
+
+import datarobot as dr
+from datarobot import CalendarFile  # type: ignore[attr-defined]
+
+from datarobotx.idp.common.hashing import get_hash
+
+
+def _find_existing_calendar(calendar_token: str) -> str:
+    for calendar in CalendarFile.list():
+        if calendar_token in str(calendar.name):
+            return str(calendar.id)
+    raise KeyError("No matching calendar found")
+
+
+def get_or_create_calendar_dataset_from_country_code(
+    endpoint: str,
+    token: str,
+    name: str,
+    country_code: str,
+    start_date: Union[str, dt.datetime],
+    end_date: Union[str, dt.datetime],
+) -> str:
+    """Create or retrieve a calendar dataset from a country code.
+
+    Parameters
+    ----------
+    name : str
+        Name for the calendar
+    country_code : str
+        The country code to use for the calendar.
+    start_date : str | datetime
+        The earliest date to include in the generated calendar.
+        If a string, it must be in the format 'YYYY-MM-DD'.
+    end_date : str | datetime
+        The latest date to include in the generated calendar.
+        If a string, it must be in the format 'YYYY-MM-DD'.
+
+    Returns
+    -------
+    str
+        The ID of the created or retrieved calendar.
+    """
+    dr.Client(token=token, endpoint=endpoint)  # type: ignore
+
+    calendar_token = get_hash(name, country_code, start_date, end_date)
```

Comment:
does our hasher work with dt.datetime objects?

Reply:
looks like we're not testing that case

Reply:
No it does not work with datetime objects. I left the option of both in while we decide if we want to support